backends-common/redis/src/main/java/org/apache/james/backends/redis/RedisConfiguration.scala

```diff
@@ -42,15 +52,24 @@ object RedisConfiguration {
 
   def from(config: Configuration): RedisConfiguration = {
     val redisConfiguration: RedisConfiguration = config.getString("redis.topology", STANDALONE_TOPOLOGY) match {
-      case STANDALONE_TOPOLOGY => StandaloneRedisConfiguration.from(config)
+      case STANDALONE_TOPOLOGY => checkTLS(config, () => TLSStandaloneRedisConfiguration.from(config), () => StandaloneRedisConfiguration.from(config))
       case CLUSTER_TOPOLOGY => ClusterRedisConfiguration.from(config)
-      case MASTER_REPLICA_TOPOLOGY => MasterReplicaRedisConfiguration.from(config)
+      case MASTER_REPLICA_TOPOLOGY => checkTLS(config, () => TLSMasterReplicaRedisConfiguration.from(config), () => MasterReplicaRedisConfiguration.from(config))
+      case SENTINEL_TOPOLOGY => checkTLS(config, () => TLSSentinelRedisConfiguration.from(config), () => SentinelRedisConfiguration.from(config))
       case _ => throw new IllegalArgumentException("Invalid topology")
     }
 
     LOGGER.info(s"Configured Redis with: ${redisConfiguration.asString}")
     redisConfiguration
   }
+
+  def checkTLS(config: Configuration, f1: () => RedisConfiguration, f2: () => RedisConfiguration): RedisConfiguration = {
+    if (config.getString(REDIS_URL_PROPERTY_NAME) != null && config.getString(REDIS_URL_PROPERTY_NAME).contains("rediss")) {
```

Comment:
Please rely on url parsing. 

The obvious `redis://rediss:35425` edge case or even `redis://rediss:pass@rerigeh:3445` would break.